crates/red_knot_python_semantic/src/types.rs

```diff
@@ -824,6 +813,96 @@ impl<'db> From<&Type<'db>> for Type<'db> {
     }
 }
 
+/// Non-exhaustive enumeration of known classes (e.g. `builtins.int`, `typing.Any`, ...) to allow
+/// for easier syntax when interacting with very common classes.
+///
+/// Feel free to expand this enum if you ever find yourself using the same class in multiple
+/// places.
+/// Note: good candidates are any classes in `[crate::stdlib::CoreStdlibModule]`
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum KnownClass {
+    // Builtins
+    Bool,
+    Object,
+    Bytes,
+    Int,
+    Float,
+    Str,
+    List,
+    Tuple,
+    Set,
+    Dict,
+    // Typing
+    Any, // Not currently used, included to show support for non-builtins
+}
+
+impl<'db> KnownClass {
+    pub const fn as_str(&self) -> &'static str {
+        match self {
+            Self::Bool => "bool",
+            Self::Object => "object",
+            Self::Bytes => "bytes",
+            Self::Tuple => "tuple",
+            Self::Int => "int",
+            Self::Float => "float",
+            Self::Str => "str",
+            Self::Set => "set",
+            Self::Dict => "dict",
+            Self::List => "list",
+            Self::Any => "Any",
+        }
+    }
+
+    pub fn to_instance(&self, db: &'db dyn Db) -> Type<'db> {
+        self.to_class(db).to_instance(db)
+    }
+
+    pub fn to_class(&self, db: &'db dyn Db) -> Type<'db> {
+        match self {
+            Self::Any => types_symbol_ty(db, self.as_str()),
+            _ => builtins_symbol_ty(db, self.as_str()),
+        }
+    }
+
+    pub fn maybe_from_module(module: &Module, class_name: &str) -> Option<Self> {
+        let candidate = Self::from_name(class_name)?;
+        if candidate.check_module(module) {
+            Some(candidate)
+        } else {
+            None
+        }
+    }
+
+    pub fn from_name(name: &str) -> Option<Self> {
+        match name {
+            "bool" => Some(Self::Bool),
+            "object" => Some(Self::Object),
+            "bytes" => Some(Self::Bytes),
+            "tuple" => Some(Self::Tuple),
+            "int" => Some(Self::Int),
+            "float" => Some(Self::Float),
+            "str" => Some(Self::Str),
+            "set" => Some(Self::Set),
+            "dict" => Some(Self::Dict),
+            "list" => Some(Self::List),
+            "Any" => Some(Self::Any),
+            _ => None,
+        }
```

Comment:
I wish Rust had some way of enforcing that the r.h.s. of a match statement is meant to be exhaustive :/ I guess that's what you were going for with your `EnumIter` solution from `strum`, but I agree with Dhruv that it's probably not really worth an extra dependency.

If this enum continues to grow and this becomes a maintenance problem, I guess we could consider writing a macro to generate this method, which would allow us to ensure that the r.h.s. of the `match` is exhaustive. But that comes with its own tradeoffs, of course.

Reply:
I agree, and indeed this is where my mind went first when I introduced `strum`: make sure that extending the enum is as painless as possible. I'm dropping a `// Note` in there for future reference. Also found a stackoverflow exploring exactly this limitation.

Reply:
If that would be more manageable, one option we would have for the maintainability of mapping str to KnownClass would be to only add a dependency for tests, which would be used to ensure we didn't miss a case.
Don't know if that's more acceptable, but it would ensure that we never miss a case without adding dependencies to the production code

Reply:
Let's land the PR for now. Maybe we can create an issue to discuss this further and see what the other red-knot maintainers think :-)